File: service/src/main/java/api/AccountService.java
package api;

import com.fasterxml.jackson.annotation.JsonProperty;
import core.PasswordReset;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/account")
public interface AccountService {

  @POST
  @Path("/request/verification")
  @Consumes(MediaType.APPLICATION_JSON)
  Response requestNewVerification(Email email);

  @POST
  @Path("/verification/")
  @Consumes(MediaType.APPLICATION_JSON)
  Response validateEmail(TokenString token);

  @POST
  @Path("/request/password_reset/")
  @Consumes(MediaType.APPLICATION_JSON)
  Response requestPasswordReset(Email email);

  @POST
  @Path("/password_reset")
  @Consumes(MediaType.APPLICATION_JSON)
  Response resetPassword(PasswordReset passwordReset);

  class Email {
    private final String email;

    public Email(@JsonProperty("email") String email) {
      this.email = email;
    }

    public String getEmail() {
      return email;
    }
  }

  class TokenString {
    private final String token;

    public TokenString(@JsonProperty("token") String token) {
      this.token = token;
    }

    public String getToken() {
      return token;
    }
  }
}

File: webapp/src/service/accountService.ts
import {Requests} from "./requests"
import {JSON_HEADER} from "./headers"
import {PasswordReset} from "../types";

export class AccountService {

    static async requestNewVerification(email: string): Promise<Response> {
        const data: string = JSON.stringify({"email": email});

        return await Requests.sendRequest("account/request/verification", {
            method: "POST",
            headers: JSON_HEADER,
            body: data
        }, false);
    }

    static async validateAccount(token: string): Promise<Response> {
        const data: string = JSON.stringify({"token": token});

        return await Requests.sendRequest("account/verification", {
            method: "POST",
            headers: JSON_HEADER,
            body: data
        }, false);
    }

    static async requestPasswordReset(email: string): Promise<Response> {
        const data: string = JSON.stringify({"email": email});

        return await Requests.sendRequest("account/request/password_reset", {
            method: "POST",
            headers: JSON_HEADER,
            body: data
        }, false);
    }

    static async resetPassword(resetPassword: PasswordReset): Promise<Response> {
        const data: string = JSON.stringify({
            "token": resetPassword.token,
            "newPassword": resetPassword.newPassword
        });

        return await Requests.sendRequest("account/password_reset", {
            method: "POST",
            headers: JSON_HEADER,
            body: data
        }, false);
    }
}